common/persistence/visibility_single_manager_test.go

```diff
@@ -95,3 +96,946 @@ func TestGetName(t *testing.T) {
+			}
+		})
+	}
+}
+
+func TestGetSearchAttributes(t *testing.T) {
+	tests := map[string]struct {
+		input          *map[string]interface{}
+		expectedOutput *types.SearchAttributes
+		expectedError  error
+	}{
+		"Case1: error case": {
+			input: &map[string]interface{}{
+				"CustomStringField": make(chan int),
+			},
+			expectedError: fmt.Errorf("error"),
+		},
+		"Case2: normal case": {
+			input:         &map[string]interface{}{},
+			expectedError: nil,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockVisibilityStore := NewMockVisibilityStore(ctrl)
+			visibilityManager := NewVisibilityManagerImpl(mockVisibilityStore, log.NewNoop())
+			visibilityManagerImpl := visibilityManager.(*visibilityManagerImpl)
+
+			actualOutput, actualErr := visibilityManagerImpl.getSearchAttributes(*test.input)
+			if test.expectedError != nil {
+				assert.Error(t, actualErr)
+			} else {
+				assert.NoError(t, actualErr)
+				assert.NotNil(t, actualOutput)
+			}
+		})
+	}
+}
+
+func TestConvertVisibilityWorkflowExecutionInfo(t *testing.T) {
+	testExecutionTime := int64(0)
+	testStartTime := time.Now()
+	testResultTime := testStartTime.UnixNano()
+
+	tests := map[string]struct {
+		input          *InternalVisibilityWorkflowExecutionInfo
+		expectedOutput *types.WorkflowExecutionInfo
+	}{
+		"Case1: error case": {
+			input: &InternalVisibilityWorkflowExecutionInfo{
+				SearchAttributes: map[string]interface{}{
+					"CustomStringField": make(chan int),
+				},
+				ExecutionTime: time.UnixMilli(int64(testExecutionTime)),
+				StartTime:     testStartTime,
+			},
+			expectedOutput: &types.WorkflowExecutionInfo{
+				ExecutionTime: &testResultTime,
+			},
+		},
+		"Case2: normal case with Execution Time is 0": {
+			input: &InternalVisibilityWorkflowExecutionInfo{
+				ExecutionTime: time.UnixMilli(int64(testExecutionTime)),
+				StartTime:     testStartTime,
+			},
+			expectedOutput: &types.WorkflowExecutionInfo{
+				ExecutionTime: &testResultTime,
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockVisibilityStore := NewMockVisibilityStore(ctrl)
+			visibilityManager := NewVisibilityManagerImpl(mockVisibilityStore, log.NewNoop())
+			visibilityManagerImpl := visibilityManager.(*visibilityManagerImpl)
+
+			actualOutput := visibilityManagerImpl.convertVisibilityWorkflowExecutionInfo(test.input)
+			assert.Equal(t, *test.expectedOutput.ExecutionTime, *actualOutput.ExecutionTime)
+		})
+	}
+}
+
+func TestToInternalListWorkflowExecutionsRequest(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockVisibilityStore := NewMockVisibilityStore(ctrl)
+	visibilityManager := NewVisibilityManagerImpl(mockVisibilityStore, log.NewNoop())
+	visibilityManagerImpl := visibilityManager.(*visibilityManagerImpl)
+
+	assert.Nil(t, visibilityManagerImpl.toInternalListWorkflowExecutionsRequest(nil))
```

Comment:
nit: this is already covered by other unit tests targeting public funcs. unless you wanted to cover specific edge cases, you don't need to add a separate test func targeting private funcs.

Reply:
Yes, I am testing an edge case with an input is nil.